app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/DecompressActivity.kt

```diff
@@ -141,6 +146,11 @@ class DecompressActivity : SimpleActivity() {
                         continue
                     }
 
+                    val isVulnerableForZipPathTraversal = !File(newPath).canonicalPath.startsWith(parent)
+                    if (isVulnerableForZipPathTraversal) {
+                        continue
+                    }
```

Comment:
this is not really a part of SDK 34 migration, but wanted to add a check for other pre 34 SDKs. 

please see the change for SDK 34 here [Android 14: Zip Path Traversal behavior changes](https://developer.android.com/about/versions/14/behavior-changes-14#zip-path-traversal
)